apps/user-portal/src/pages/consents.tsx

```diff
@@ -43,37 +50,63 @@ export class ConsentsPage extends React.Component<any, any> {
             editingConsent: createEmptyConsent(),
             consentReceipt: createEmptyConsentReceipt(),
             showConsentEditModal: false,
-            showConsentRevokeModal: false
+            showConsentRevokeModal: false,
+            revokedClaimIds: []
         };
     }
 
     public componentWillMount() {
-        getConsents(ConsentState.ACTIVE).then((response) => {
-            this.setState({ consents: response });
-        });
+        this.updateConsents(ConsentState.ACTIVE);
     }
 
+    /**
+     * Handles the tab onclick event. The active tab index is also
+     * stored in the state and the list of consents is updated
+     * according to the selected tab.
+     * @param e event
+     * @param {any} activeIndex active tab index
+     */
     public handleConsentTabChange = (e, { activeIndex }) => {
         this.setState({ activeIndex }, () => {
-            let consentState = ConsentState.ACTIVE;
-            switch (activeIndex) {
-                case 0:
-                    consentState = ConsentState.ACTIVE;
-                    break;
-                case 1:
-                    consentState = ConsentState.REVOKED;
-                    break;
-                default:
-                    consentState = ConsentState.ACTIVE;
-                    break;
-            }
+            const consentState: ConsentState = this.getConsentState(activeIndex);
+            this.updateConsents(consentState);
+        });
+    }
 
-            getConsents(consentState).then((response) => {
-                this.setState({ consents: response });
-            });
+    /**
+     * Fetches the consents list from the API and updates the state.
+     * @param {ConsentState} state consent state ex: ACTIVE, REVOKED
+     */
+    public updateConsents = (state: ConsentState) => {
+        getConsents(state).then((response) => {
+            this.setState({ consents: response });
         });
     }
 
+    /**
+     * Retrieves the consent state corresponding to the active
+     * @param {number} tabIndex active tab index
+     * @return {ConsentState} consent state ex: ACTIVE, REVOKED
+     */
+    public getConsentState = (tabIndex: number) => {
+        let consentState = ConsentState.ACTIVE;
+        switch (tabIndex) {
```

Comment:
Since we have only ACTIVE state now I think we can remove the switch statement here